src/Aether/Devices/Drivers/Sgp4x.cs

```diff
@@ -0,0 +1,210 @@
+            }
+
+            // Read serial number into array excluding CRC bytes
+            byte[] serialNumber = new byte[6];
+
+            serialNumber[0] = serialNumberWithCRC[0];
+            serialNumber[1] = serialNumberWithCRC[1];
+            serialNumber[2] = serialNumberWithCRC[3];
+            serialNumber[3] = serialNumberWithCRC[4];
+            serialNumber[4] = serialNumberWithCRC[6];
+            serialNumber[5] = serialNumberWithCRC[7];
+
+            return serialNumber;
+        }
+
+        /// <summary>
+        /// Runs a self test on the device.
+        /// </summary>
+        /// <returns>True if all tests passed. False if one or more tests failed.</returns>
+        public bool RunSelfTest()
+        {
+            Span<byte> runSelfTestCommand = stackalloc byte[2] { 0x28, 0x0E };
+            Span<byte> testResultWithCRC = stackalloc byte[3];
+
+            // Write run self test command
+            _device.Write(runSelfTestCommand);
+
+            Thread.Sleep(320);
+
+            // Read test result data
+            _device.Read(testResultWithCRC);
+
+            ushort? readValue;
+            readValue = Sensirion.ReadUInt16BigEndianAndCRC8(testResultWithCRC);
+
+            if (readValue is null)
+            {
+                throw new IOException("CRC for receved data is invalid");
+            }
+            // Check result status (Most significant byte, ignore least significant non-crc byte)
+            // 0xD4 = All tests passed
+            // 0x4B = One or more test failed
+            return testResultWithCRC[0] == 0xD4;
+        }
+
+        /// <summary>
+        /// Gets the raw VOC measurements from the device.
+        /// </summary>
+        /// <param name="relativeHumidityValue">The relative humidity value expressed as a percentage.</param>
+        /// <param name="temperatureValue">The temperature value expressed in degrees celsius.</param>
+        /// <returns>The raw VOC measurement value from the sensor. If an error occurred, it will be <see langword="null"/>.</returns>
+        /// <remarks>If default relative humidity and temperature values are supplied, humidity compensation will be disabled.</remarks>
+        public VolatileOrganicCompoundIndex? GetVOCRawMeasure(RelativeHumidity? relativeHumidityValue = null, Temperature? temperatureValue = null)
+        {
+            double rhValue = 50;
+            double tempValue = 25;
+
+            if(relativeHumidityValue is not null)
+            {
+                rhValue = relativeHumidityValue.Value.Value;
+
+                if (rhValue < 0 || rhValue > 100)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(relativeHumidityValue), relativeHumidityValue, "A relative humidity percentage value must be between 0 and 100");
+                }
+            }
+            
+            if (temperatureValue is not null)
+            {
+                tempValue = temperatureValue.Value.Value;
```

Comment:
`Value` might be in any number of units; need to the specific one you need.

```suggestion
                tempValue = temperatureValue.Value.DegreesFahrenheit;
```